File: web_full_stack/ruby/lib/reverb_client.rb
require 'httparty'

class ReverbClient
  HEADERS = {
    'Accept' => 'application/json',
    'Accept-Version' => '3.0',
    'Content-Type' => 'application/json'
  }

  attr_reader :base_uri

  def initialize(base_uri: 'https://api.reverb.com/api')
    @base_uri = base_uri
  end

  def listings(per_page: 10, currency: 'EUR', shipping_region: 'FR')
    pp currency
    pp shipping_region
    get('/listings/all', {per_page: per_page}, { 'X-Display-Currency' => currency,  'X-Shipping-Region' => shipping_region })['listings']
  end

  def categories
    get('/categories/flat')['categories']
  end

  private

  def get(path, params={}, extra_headers = {})
    # merges the extra headers with HEADERS
    JSON.parse(HTTParty.get(base_uri + path, headers: HEADERS.merge(extra_headers), query: params).body)
  end
end

File: web_full_stack/python/reverb_client.py
import requests

class ReverbClient:
  HEADERS = {
    'Accept': 'application/json',
    'Accept-Version': '3.0',
    'Content-Type': 'application/json'
  }

  def __init__(self, base_uri='https://api.reverb.com/api'):
    self._base_uri = base_uri

  def listings(self, per_page=10):
    return self._get('/listings/all', { 'per_page': per_page })['listings']

  def categories(self):
    return self._get('/categories/flat')['categories']

  def _get(self, path, params={}):
    return requests.get(
      self._base_uri + path,
      headers=self.HEADERS,
      params=params
    ).json()
